### PR TITLE
notification clear all plugin

### PR DESCRIPTION
**Pull request by sfeng88**
- [x] joeferner
- [x] sfeng88
- [ ] mwizeman joeybrk372 jharwig

Testing Instructions:
- Clicking on the trash can button should clear all notifications in the dashboard card. Non-system notifications should not reappear after a browser refresh

Points of Regression: N/A

CHANGELOG
Added: Notification clear all plugin for the notification dashboard card

### CODE REVIEW

**Comment by joeferner** [comment]
@sfeng88 should this just be part of the notifications instead of a new plugin? If someone doesn't want it we could add a configuration toggle later. BTW I would like this feature.

File: web/plugins/notifications-clear-all/src/main/resources/org/visallo/web/notifications/plugin.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+define(['public/v1/api'], function(visallo) {
+    'use strict';
+
+    visallo.registry.registerExtension('org.visallo.dashboard.toolbar.item', {
```
Comment (joeybrk372) [inline web/plugins/notifications-clear-all/src/main/resources/org/visallo/web/notifications/plugin.js]: I agree with Joe. You can register this extension at the top of `notifications.js` and then listen for the event directly in the component without having to send an extra request to get all notifications

File: web/plugins/notifications-clear-all/src/main/resources/org/visallo/web/notifications/plugin.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        identifier: 'org-viasllo-notification-clear-all',
```
Comment (joeybrk372) [inline web/plugins/notifications-clear-all/src/main/resources/org/visallo/web/notifications/plugin.js]: typo in visallo

**Comment by jharwig** [comment]
@joeferner  Yes, I wasn't imagining this would be a (jar) plugin, just that the built-in notifications used the dashboard tools extension point to add the clear button. (what @joeybrk372  said)

**Comment by sfeng88** [comment]
@joeferner @jharwig @joeybrk372  Fixed, please take another look at the PR

**Pull request merged**